### PR TITLE
Fix incorrect `Parameter` range  for `*args` and `**kwargs` 

### LINKED ISSUE #10282: `Parameter` range is incorrect for `*args` and `**kwargs`
I'm working on a fix for the `flake8-unused-arguments` (ARG) rules that would delete them.

While working on this, I noticed that the `*args` and `**kwargs` `Parameter.range` does not include the `*`. 
I think that `Parameter.range` should include `*` or `**`, leaving `Parameter.name.range` for the actual name range if needed.

```python
def f(*args, **kwargs): pass
#      ~~~~    ~~~~~~    <-- actual range
#     ^^^^^  ^^^^^^^^    <-- expected range
```

<details>
<summary>See the currently generated AST</summary>

https://github.com/astral-sh/ruff/blob/7b4a73d421c130430883bb944a285f096e3aa9a5/crates/ruff_python_parser/src/snapshots/ruff_python_parser__function__tests__function_pos_and_kw_only_args_with_defaults_and_varargs_and_kwargs.snap#L58-L67

Here I would expect the `Parameter` range to be 1 longer than the `Identifier` range.

https://github.com/astral-sh/ruff/blob/7b4a73d421c130430883bb944a285f096e3aa9a5/crates/ruff_python_parser/src/snapshots/ruff_python_parser__function__tests__function_pos_and_kw_only_args_with_defaults_and_varargs_and_kwargs.snap#L124-L134

And here I would expect the `Parameter` range to be 2 longer than the `Identifier` range.
</details>

If this is indeed considered a bug, I have a PR ready to fix it.

### PR DESCRIPTION
**Pull request by GtrMo**
## Summary

Fix #10282 

This PR updates the Python grammar to include the `*` character in `*args` `**kwargs` in the range of the `Parameter`
```
def f(*args, **kwargs): pass
#      ~~~~    ~~~~~~    <-- range before the PR
#     ^^^^^  ^^^^^^^^    <-- range after
```

The invalid syntax `def f(*, **kwargs): ...` is also now correctly reported.

## Test Plan

Test cases were added to `function.rs`.

### CODE REVIEW

File: crates/ruff_python_parser/src/python.rs
Comment (GtrMo) [inline crates/ruff_python_parser/src/python.rs:1]: Is the diff supposed to be that big? I used the same version of `lalrpop` (0.20.0) to update this file

Comment (MichaReiser) [reply crates/ruff_python_parser/src/python.rs:1]: Yeah, lalrpop loves to generate large diffs :(

**Comment by MichaReiser** [comment]
Changing the ranges drips over the formatter. There are multiple places where the formatter goes back to introspecting the source to identify the comment placement and I fear that the code has now been written in a way that assumes the incorrect `*arg` and `**kwargs` ranges. 

I've been able to identify a problematic example:

```python
(
    lambda # comment 1
    * # comment 2
    x: # comment 3
    x
)
```

But I haven't yet been successful to narrow it down where the range handling is happening. I suspect that it is related to https://github.com/astral-sh/ruff/issues/10281 (CC: @dhruvmanila) but haven't been able to confirm it yet.

**Review by MichaReiser — APPROVED** [review]
Looks good to me. Would you mind rebasing and regenerating the parser (after rebase)?

**Comment by charliermarsh** [comment]
I can rebase, regen, and merge this later.

**Comment by GtrMo** [comment]
Thanks for the formatter fix! I rebased and regenerated the parser

**Pull request merged**